src/ui.rs

```diff
@@ -105,13 +126,20 @@ pub fn event_action(state: &mut State, event: &Event) -> Action {
                 state.render_infobar = !state.render_infobar;
                 state.process_action(Action::ReRender)
             }
-
-            W | PageUp => state.process_action(Action::SkipForward),
-            B | PageDown => state.process_action(Action::SkipBack),
-            Z => state.process_action(Action::ToggleFit),
+            W | PageUp => Action::SkipForward,
+            B | PageDown => Action::SkipBack,
+            Z => Action::ToggleFit,
+            Home => Action::First,
+            End => Action::Last,
             Period => state.last_action.clone(),
-            Home => state.process_action(Action::First),
-            End => state.process_action(Action::Last),
+            Semicolon => {
+                if state.left_shift || state.right_shift {
```

Comment:
Really not sure this holds true on non-american keyboards. Shift semicolon does not always equal colon. Please see if there's another way to do this. Maybe a virtual keycode if you can get them working.